music_player2/docs/ch/TECHNICAL.md

```diff
@@ -0,0 +1,422 @@
+├─────────────────────────────────────────┤
+│             服务层                      │  ← 音频控制、文件管理、Wi-Fi管理
+├─────────────────────────────────────────┤
+│        系统抽象层 (NuttX)                │  ← 音频驱动、文件系统、网络栈
+├─────────────────────────────────────────┤
+│           硬件抽象层                     │  ← 音频硬件、存储设备、网络硬件
+└─────────────────────────────────────────┘
+```
+
+### 数据流图
+
+系统的数据流主要包括 3 个主要路径：
+
+#### 1. 应用启动流
+```
+main() → splash_screen_create() → 启动动画 → app_create() → 主界面
+   ↓
+资源初始化 → 字体加载 → 配置读取 → UI组件创建
+```
+
+#### 2. 用户交互流
+```
+用户操作 → LVGL 9.x 事件 → 事件处理函数 → 状态更新 → UI刷新
+    ↓
+音频控制 → 播放列表更新 → 界面反馈
+```
+
+#### 3. 模块通信流
+```
+主界面 ↔ 播放列表管理器 ↔ 音频控制器 ↔ 状态管理
+   ↓              ↓              ↓
+UI更新        列表显示        播放控制
+```
+
+### 主要数据结构
+
+#### 运行时上下文
+```c
+struct ctx_s {
+    bool resource_healthy_check;          // 资源健康检查标志
+    album_info_t* current_album;          // 当前播放的专辑信息
+    
+    // 播放状态
+    uint16_t volume;                      // 当前音量 (0-100)
+    play_status_t play_status_prev;       // 上一次播放状态
+    play_status_t play_status;            // 当前播放状态
+    uint64_t current_time;                // 当前播放时间 (毫秒)
+    
+    // 定时器管理
+    struct {
+        lv_timer_t* volume_bar_countdown;      // 音量条自动隐藏倒计时
+        lv_timer_t* playback_progress_update;  // 播放进度更新定时器
+        lv_timer_t* refresh_date_time;         // 时间日期刷新定时器
+    } timers;
+    
+    audioctl_s* audioctl;                 // 音频控制句柄
+};
+```
+
+#### 资源管理结构
+```c
+struct resource_s {
+    // UI组件管理
+    struct {
+        lv_obj_t* time;                    // 时间显示标签
+        lv_obj_t* date;                    // 日期显示标签
+        lv_obj_t* player_group;            // 播放器控件组
+        lv_obj_t* album_cover_container;   // 圆形封面容器
+        lv_obj_t* volume_bar;              // 音量条组件
+        // ... 更多UI组件
+    } ui;
+    
+    // 字体资源管理
+    struct {
+        struct { const lv_font_t* normal; } size_16;
+        struct { const lv_font_t* bold; } size_22;
+        struct { const lv_font_t* normal; } size_24;
+        // ... 更多字体
+    } fonts;
+    
+    // 音乐数据
+    album_info_t* albums;                  // 专辑信息数组
+    uint8_t album_count;                   // 专辑总数
+};
+```
+
+### 状态机设计
+
+#### 播放状态定义
+```c
+typedef enum {
+    PLAY_STATUS_STOP,    // 停止状态
+    PLAY_STATUS_PLAY,    // 播放状态
+    PLAY_STATUS_PAUSE,   // 暂停状态
+} play_status_t;
+```
+
+#### 状态转换表
+```c
+typedef struct {
+    play_status_t from_state;     // 源状态
+    play_status_t to_state;       // 目标状态
+    void (*action)(void);         // 转换动作
+} state_transition_t;
+
+// 状态转换表
+static const state_transition_t transitions[] = {
+    {PLAY_STATUS_STOP,  PLAY_STATUS_PLAY,  action_start_playback},
+    {PLAY_STATUS_PLAY,  PLAY_STATUS_PAUSE, action_pause_playback},
+    {PLAY_STATUS_PAUSE, PLAY_STATUS_PLAY,  action_resume_playback},
+    {PLAY_STATUS_PLAY,  PLAY_STATUS_STOP,  action_stop_playback},
+    {PLAY_STATUS_PAUSE, PLAY_STATUS_STOP,  action_stop_playback},
+};
+```
+
```

Comment:
This code does not appear in the main code, please confirm the accuracy of the technical documentation.